packages/doenetml-worker-rust/lib-doenetml-core/src/core/dependency.rs

```diff
@@ -396,3 +585,35 @@ fn get_children_with_parent_including_from_extend_source(
 
     children_vec
 }
+
+/// Return the attribute children for `attribute`,
+/// falling back to the attribute children of any extend source if none found for the component.
+///
+/// Returns an option of a tuple with components
+/// - a vector of the attribute children found
+/// - the index of the parent where those attribute children were found.
+fn get_attribute_children_with_parent_falling_back_to_extend_source(
+    components: &Vec<Rc<RefCell<ComponentEnum>>>,
+    component_idx: ComponentIdx,
+    attribute: AttributeName,
+) -> Option<(Vec<ComponentPointerTextOrMacro>, ComponentIdx)> {
+    let component = components[component_idx].borrow();
+
+    let attribute_children_option = component.get_attribute_children_for_attribute(attribute);
+
+    match attribute_children_option {
+        Some(attribute_children) => {
+            // if we found the attribute but there are no children, then recurse to extend source, if it exists
+            if attribute_children.is_empty() {
+                if let Some(&ExtendSource::Component(source_idx)) = component.get_extend() {
+                    return get_attribute_children_with_parent_falling_back_to_extend_source(
+                        components, source_idx, attribute,
+                    );
+                }
+            }
+
+            Some((attribute_children.clone(), component_idx))
+        }
+        None => None,
```

Comment:
This is what `.map` does.

Reply:
I couldn't get `.map` to work because of the recursion that returns an option, but I got `.and_then` to work. I'm slowly figuring out how these functions work!